R/guide-colorbar.r

```diff
@@ -109,8 +117,15 @@ guide_colourbar <- function(
   nbin = 20,
   raster = TRUE,
 
+  # frame
+  frame.colour = NULL,
+  frame.linewidth = 0.5,
+  frame.linetype = 1,
```

Comment:
Have we used `linewidth` and `linetype` in other functions?

Reply:
`linetype` is a standard aesthetic, so I think that's fine. `linewidth` is usually called `size` throughout the rest of ggplot2. If you prefer `size` I'm happy to make the change, it's just that `frame.size` doesn't sound like it controls the thickness of the frame.

Reply:
I don't have any particularly strong feelings here; I don't really understand how these options are supposed to fit in with the theme system.

Reply:
Not sure. The question is whether every possible option to a guide function needs a theme equivalent. My personal opinion is no. I think there is a lot of room for improvement for making guide functions more flexible, and possibly providing different ones or frameworks for extending them, and it's impossible to add new theme components every time somebody makes some minor tweak to some guide function. Already, there are things the guide functions can do that the theme can't control, such as the `draw.ulim` option for `guide_colorbar()`.

Reply:
Agreed. I'm happy to merge this PR; I just wanted to make sure we at least talked over the issues.